File: atcoder/abs/arc065_a.py
import re

def solve0(S):
    r = re.compile(r'^(dreamer|eraser|dream|erase)*$')
    return r.match(S)

def solve(S):
    words = "dreamer", "eraser", "dream", "erase"
    while S:
        for word in words:
            if S.endswith(word):
                S = S[:-len(word)]
                break
        else:
            return False
    return True

S = input()

if solve(S):
    print("YES")
else:
    print("NO")

File: atcoder-go/practice/arc065_a.go
package main

import (
	"fmt"
	_ "strings"
)

func main() {
	var s string
	fmt.Scan(&s)
	var words = []string{"dream", "dreamer", "erase", "eraser"}
	i := len(s)
	var result bool
	for i > 0 {
		result = false
		for _, w := range words {
			l := len(w)
			j := i - l
			if j >= 0 && s[j:i] == w {
				result = true
				i = j
				break
			}
		}
		if !result {
			break
		}
	}
	if result {
		fmt.Print("YES")
	} else {
		fmt.Print("NO")
	}
}
